hera_pspec/tests/test_uvpspec.py

```diff
@@ -532,48 +542,41 @@ def test_combine_uvpspec(self):
                        np.ones(10, dtype=np.float64))))
         nt.assert_true(np.all(np.isclose(out.get_integrations(key), 
                        190 * np.ones(10, dtype=np.float64), atol=5, rtol=2)))
-
-        # test multiple non-overlapping data axes
-        uvp2.freq_array[0] = 0.0
-        nt.assert_raises(AssertionError, uvpspec.combine_uvpspec, [uvp1, uvp2])
-
-        # test partial data overlap failure
-        uvp2 = testing.uvpspec_from_data(uvd, [(37, 38), (38, 39), (53, 54)],
-                                         spw_ranges=[(20, 30), (60, 90)],
-                                         beam=beam)
-        nt.assert_raises(AssertionError, uvpspec.combine_uvpspec, [uvp1, uvp2])
-        uvp2 = testing.uvpspec_from_data(uvd, bls,
-                                         spw_ranges=[(20, 30), (60, 105)],
-                                         beam=beam)
-        nt.assert_raises(AssertionError, uvpspec.combine_uvpspec, [uvp1, uvp2])
-        uvp2 = copy.deepcopy(uvp1)
-        uvp2.polpair_array[0] = 1414
-        uvp2 = uvpspec.combine_uvpspec([uvp1, uvp2], verbose=False)
-        nt.assert_raises(AssertionError, uvpspec.combine_uvpspec, [uvp1, uvp2])
+        # optionals
+        for spw in out.spw_array:
+            ndlys = out.get_spw_ranges(spw)[0][-1]
+            assert out.cov_array[spw].shape == (30, ndlys, ndlys, 2)
+            assert out.stats_array['noise_err'][spw].shape == (30, ndlys, 2)
+            assert out.window_function_array[spw].shape == (30, ndlys, ndlys, 2)
```

Comment:
Should we check the cov_model is also as expected?

Reply:
sure thing